service/cluster/cluster.go

```diff
@@ -56,15 +57,20 @@ type Options struct {
 	// possible for other nodes to join the cluster.
 	EnableClustering bool
 
-	NodeName         string        // Name to use for this node in the cluster.
-	AdvertiseAddress string        // Address to advertise to other nodes in the cluster.
-	RejoinInterval   time.Duration // How frequently to rejoin the cluster to address split brain issues.
+	NodeName                string        // Name to use for this node in the cluster.
+	AdvertiseAddress        string        // Address to advertise to other nodes in the cluster.
+	RejoinInterval          time.Duration // How frequently to rejoin the cluster to address split brain issues.
+	ClusterMaxInitJoinPeers int           // Number of initial peers to join from the discovered set
 
 	// Function to discover peers to join. If this function is nil or returns an
 	// empty slice, no peers will be joined.
 	DiscoverPeers func() ([]string, error)
 }
 
+type RandShuffler interface {
+	Shuffle(n int, swap func(i, j int))
+}
+
```

Comment:
Is the reason we're adding this only for the tests?

I'm asking because, 'math/rand' already contains a [rand.Shuffle function](https://pkg.go.dev/math/rand#Shuffle) we can use directly with the default rand source? Then in the tests we can reliably seed it with the same number.

Reply:
yes this was only for the test. You are right, we could make it work with the seeding instead of mocking it. Let's go with your suggestion because it's less code and I don't really see any cons